TaskService/App_Start/OpenIdConnectCachingSecurityTokenProvider.cs

```diff
@@ -64,32 +46,15 @@ public IEnumerable<SecurityToken> SecurityTokens
         {
             get
             {
-                RetrieveMetadata();
-                _synclock.EnterReadLock();
-                try
-                {
-                    return _tokens;
-                }
-                finally
-                {
-                    _synclock.ExitReadLock();
-                }
+                return RetrieveMetadata().Result.SigningTokens;
```

Comment:
We're still calling .Result on an async method. Is that the best way, @brentschmaltz?